python-lib/dku_utils/taints.py

```diff
@@ -0,0 +1,53 @@
+import json
+from dku_utils.access import _is_none_or_blank
+
+class Taint(dict):
+    def __init__(self, taint):
+        if not _is_none_or_blank(taint.get('key', '')):
+            self['key'] = taint.get('key', '')
+
+        if not _is_none_or_blank(taint.get('value', '')):
+            self['value'] = taint.get('value', '')
+
+        if not _is_none_or_blank(taint.get('effect', '')):
+            self['effect'] = taint.get('effect', '')
+
+    def __eq__(self, other):
+        return self.get('key', '') == other.get('key', '') and self.get('value', '') == other.get('value', '') and self.get('effect', '') == other.get('effect', '')
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash((self.get('key', ''), self.get('value', ''), self.get('effect', '')))
+
+class Toleration(Taint):
+    def __init__(self, taint):
+        super().__init__(taint)
+
+        if self.get('value', ''):
+            self['operator'] = 'Equal'
+        else:
+            self['operator'] = 'Exists'
+
+    def __eq__(self, other):
+        return super().__eq__(other) and self.get('operator', '') == other.get('operator', '')
+
+
+    def __hash__(self):
+        return hash((super().__hash__(), self.get('operator', '')))
+
+    def to_dict(self):
+        return {k: v for k, v in self.items()}
+    
+    @staticmethod
+    def from_json(tolerations_json):
+        return [Toleration(tol) for tol in json.loads(tolerations_json)]
+
+    @staticmethod
+    def from_dict(raw_dicts):
+        return [Toleration(raw) for raw in raw_dicts or []]
+
+    @staticmethod
+    def to_dict(tolerations):
+        return [toleration.to_dict() for toleration in tolerations]
```

Comment:
there is an issue with this method
it is defined to be called with an argument and not called with this argument
I think the one you want to use is a native Python

Reply:
ya it turns out you're not allowed to have a static method and a class method with the same name in python.